site/src/components/layout/layout.md

```diff
@@ -81,3 +81,17 @@ svg.append('g')
       'flexDirection': 'row'
   });
 ```
+
+When building high performance charts e.g. when targeting 60fps, layout can be a source of performance problems. So in order to get the best performance, it's worth understanding some of its internals.
+
+At a high level, when a node is laid-out -
+
+* if it has not previously been laid-out it, its inner dimensions are measured. These are then combined with its `layout-style` and those of its descendants. After running these through the `css-layout` algorithm, the results are then applied node-by-node to the subtree.
+
+* if it has previously been laid-out, its inner dimensions are not be measured. Instead the `layout-width`, `layout-height`, `layout-x` and `layout-y` attributes set on it by the previous layout will be used as the basis for the algorithm. These are combined with the `layout-style` attributes of the node and its descendants then applied to the subtree as above.
```

Comment:
How does this effect re-sizing? if layout is invoked, then the browser resized, the inner dimensions could potentially change, however, according to this rule, the inner dimensions will not be re-measured.